cmd/lncli/commands.go

```diff
@@ -538,6 +569,190 @@ func showRoutingTable(ctx *cli.Context) error {
 		return err
 	}
 
-	printRespJson(resp)
+	// TODO(mkl): maybe it is better to print output directly omitting
+	// conversion to RoutingTable. This part is not performance critical so
+	// I think it is ok because it enables code reuse
+	r := rt.NewRoutingTable()
+	for _, channel := range resp.Channels {
+		r.AddChannel(
+			graph.NewID(channel.Id1),
+			graph.NewID(channel.Id2),
+			graph.NewEdgeID(channel.EdgeID),
+			&rt.ChannelInfo{channel.Capacity, channel.Weight},
+		)
+	}
+
+	req2 := &lnrpc.GetInfoRequest{}
+	resp2, err := client.GetInfo(ctxb, req2)
+	if err != nil {
+		return err
+	}
+	self, _ := hex.DecodeString(resp2.LightningId)
+
+	typ := ctx.String("type")
+	viz := ctx.String("viz")
+	if typ != "" || viz != "" {
+		TempFile, err  := ioutil.TempFile("", "") 
+		if err != nil {
+			return err
+		}
+		var ImageFile *os.File
+		// if the type is not specified explicitly parse the filename
+		if typ == "" {
+			typ = filepath.Ext(viz)[1:]
+		} 
+		// if the filename is not specified explicitly use tempfile
+		if viz == "" {
+			ImageFile, err = TempFileWithSuffix("", "rt_", "."+typ)
+			if err != nil {
+				return err
+			}
+		} else {
+			ImageFile, err = os.Create(viz)
+			if err != nil {
+				return err
+			}
+		}
+		if _, ok := visualizer.SupportedFormatsAsMap()[typ]; !ok {
+			fmt.Printf("Format: '%v' not recognized. Use one of: %v\n", typ, visualizer.SupportedFormats())
+			return nil
+		}
+		// generate description graph by dot language
+		writeToTempFile(r, TempFile, self)
+		writeToImageFile(TempFile, ImageFile)
+		if ctx.Bool("open") {
+			if err := visualizer.Open(ImageFile); err != nil {
+				return err
+			}
+		}
+	} else if ctx.Bool("table") {
+		printRTAsTable(r, ctx.Bool("human"))
+	} else {
+		printRTAsJSON(r)
+	}
 	return nil
 }
+
+func writeToTempFile(r *rt.RoutingTable, file *os.File, self []byte) {
+	slc := []graph.ID{graph.NewID(string(self))}
+	viz := visualizer.New(r.G, slc, nil, nil)
+	viz.ApplyToNode = func(s string) string { return hex.EncodeToString([]byte(s)) }
+	viz.ApplyToEdge = func(info interface{}) string { 
+		if info, ok := info.(*rt.ChannelInfo); ok {
+			return fmt.Sprintf(`"%v"`, info.Capacity())
+		}
+		return "nil"
+	}
+	// need to call method if plan to use shortcut, autocomplete, etc
+	viz.BuildPrefixTree()
+	viz.EnableShortcut(true)
+	dot := viz.Draw()
+	file.Write([]byte(dot))
+	file.Sync()
+}
+
+func writeToImageFile(TempFile, ImageFile *os.File) {
+	visualizer.Run("neato", TempFile, ImageFile)
+	TempFile.Close()
+	os.Remove(TempFile.Name())
+	ImageFile.Close()
+}
+
+// get around a bug in the standard library, add suffix param
+func TempFileWithSuffix(dir, prefix, suffix string) (f *os.File, err error) {
+	f, err = ioutil.TempFile(dir, prefix)
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(f.Name())
+	f, err = os.Create(f.Name()+suffix)
```

Comment:
Using named returns should be avoided to maintain consistency with the rest of the codebase.